### ISSUE TITLE
[New component] Input search

### ISSUE THREAD

**Issue by geospatialem** [Esri/calcite-design-system#10186]
**monday.com sync:** #18048538324

### Description

Autocomplete from https://github.com/Esri/calcite-design-system/issues/10044 would leverage this component to adopt autocomplete functionality.

Blocked issues: #10187, #8877

### User Stories

- Dynamic searching within `combobox`
- Similar to MUI's [React autocomplete component](https://mui.com/material-ui/react-autocomplete/#search-as-you-type)

### Acceptance Criteria

- [ ] Supports autocomplete
- [ ] Provides context to results via a live region for assistive technologies

### Relevant Info

https://github.com/Esri/calcite-design-system/issues/10044

### Priority impact

impact - p2 - want for an upcoming milestone

### Esri team

Calcite (dev)

# CalciteInputSearch

- New component `calcite-input-search` in order to setup the input as a search.

```html
<calcite-input-search placeholder="Enter your region"></calcite-input-search>
```

### Properties consistent with `calcite-input-text`

- autocomplete: string;
- disabled: boolean;
- form: string;
- icon: Boolean | IconNameOrString = true; // renders search icon when true
- iconFlipRtl: boolean;
- label: string;
- loading:boolean;
- maxLength: number;
- messageOverrides: { label: string; clear: string; };;
- minLength: number;
- name: string;
- pattern: string; // not on autocomplete
- placeholder: string;
- prefixText: string;
- readOnly = false; // not on autocomplete
- required = false;
- scale: Scale;
- status: Status = "idle";
- suffixText: string;
- validationIcon: IconNameOrString | boolean;
- validationMessage: string;
- readonly validity;
- value: string;

#### Global attributes supported

- autofocus: boolean; // global attribute
- enterkeyhint: boolean; // global attribute
- inputmode: boolean; // global attribute

## Events

- calciteInputSearchTextChange: EventEmitter<void>; // Fires each time a new value is typed and committed.
- calciteInputSearchTextInput: EventEmitter<void>; // Fires each time a new value is typed.

## Methods

- selectText()
- setFocus()

## Slots

- actions-start
- actions-end

**Comment by geospatialem** [comment]
Once #10044 lands, we should discuss the above to determine future prioritization.

**Comment by DitwanP** [comment]
Spike to determine if needed to complement the autocomplete component or if no longer needed. Coordination with Aaron and Matt.

**Comment by DintaMel** [comment]
Closing due to Ditwan's findings and lack of need.

> Findings: We don't really need this component at the moment, we could want/need it farther down the line but overall a very low priority at the moment.
> 
> I'm moving this into the freezer for now.
